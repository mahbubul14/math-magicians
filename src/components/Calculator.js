import React from 'react';
import calculate from '../logic/calculate';

class Calculator extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      total: 0,
      next: null,
      operation: null,
    };

    this.handleEvent = this.handleEvent.bind(this);
  }

  handleEvent = (buttonName) => {
    this.setState((state) => calculate(state, buttonName));
  };

  render() {
    const { total, operation, next } = this.state;
    return (
      <div>
        <p className="result">
          <span>{total}</span>
          <span>{operation}</span>
          <span>{next}</span>
        </p>
        <div className="button">
          <button type="button" className="grayBtn" name="AC" onClick={(e) => this.handleEvent(e.target.name)}>AC</button>
          <button type="button" className="grayBtn" name="+/-" onClick={(e) => this.handleEvent(e.target.name)}>+/-</button>
          <button type="button" className="grayBtn" name="%" onClick={(e) => this.handleEvent(e.target.name)}>%</button>
          <button type="button" className="orangeBtn" name="รท" onClick={(e) => this.handleEvent(e.target.name)}>รท</button>
          <button type="button" className="grayBtn" name="7" onClick={(e) => this.handleEvent(e.target.name)}>7</button>
          <button type="button" className="grayBtn" name="8" onClick={(e) => this.handleEvent(e.target.name)}>8</button>
          <button type="button" className="grayBtn" name="9" onClick={(e) => this.handleEvent(e.target.name)}>9</button>
          <button type="button" className="orangeBtn" name="x" onClick={(e) => this.handleEvent(e.target.name)}>x</button>
          <button type="button" className="grayBtn" name="4" onClick={(e) => this.handleEvent(e.target.name)}>4</button>
          <button type="button" className="grayBtn" name="5" onClick={(e) => this.handleEvent(e.target.name)}>5</button>
          <button type="button" className="grayBtn" name="6" onClick={(e) => this.handleEvent(e.target.name)}>6</button>
          <button type="button" className="orangeBtn" name="-" onClick={(e) => this.handleEvent(e.target.name)}>-</button>
          <button type="button" className="grayBtn" name="1" onClick={(e) => this.handleEvent(e.target.name)}>1</button>
          <button type="button" className="grayBtn" name="2" onClick={(e) => this.handleEvent(e.target.name)}>2</button>
          <button type="button" className="grayBtn" name="3" onClick={(e) => this.handleEvent(e.target.name)}>3</button>
          <button type="button" className="orangeBtn" name="+" onClick={(e) => this.handleEvent(e.target.name)}>+</button>
          <button type="button" className="grayBtn" id="zero" name="0" onClick={(e) => this.handleEvent(e.target.name)}>0</button>
          <button type="button" className="grayBtn" name="." onClick={(e) => this.handleEvent(e.target.name)}>.</button>
          <button type="button" className="orangeBtn" name="=" onClick={(e) => this.handleEvent(e.target.name)}>=</button>
        </div>
      </div>
    );
  }
}
export default Calculator;
