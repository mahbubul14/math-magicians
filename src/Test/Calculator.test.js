import { React } from 'react';
import { act } from 'react-dom/test-utils';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

let container;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

describe('Calculator component', () => {
  it('Just Renders', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('can render and update result', () => {
    act(() => {
      ReactDOM.render(<Calculator />, container);
    });
    const numBtn = container.querySelectorAll('.grayBtn')[3];
    const result = container.querySelector('.result');
    expect(result.textContent).toBe('0');

    act(() => {
      numBtn.dispatchEvent(new MouseEvent('click', { bubbles: true }));
    });
    expect(result.textContent).toBe('7');
  });
});
