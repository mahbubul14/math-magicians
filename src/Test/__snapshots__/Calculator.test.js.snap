// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div>
  <h2
    className="cal-title"
  >
     let's do some Math!
  </h2>
  <div
    className="cal-float-right"
  >
    <p
      className="result"
    >
      <span>
        0
      </span>
      <span />
      <span />
    </p>
    <div
      className="button"
    >
      <button
        className="grayBtn"
        name="AC"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="grayBtn"
        name="+/-"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="grayBtn"
        name="%"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="orangeBtn"
        name="รท"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="grayBtn"
        name="7"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="grayBtn"
        name="8"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="grayBtn"
        name="9"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="orangeBtn"
        name="x"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="grayBtn"
        name="4"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="grayBtn"
        name="5"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="grayBtn"
        name="6"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="orangeBtn"
        name="-"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="grayBtn"
        name="1"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="grayBtn"
        name="2"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="grayBtn"
        name="3"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="orangeBtn"
        name="+"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="grayBtn"
        id="zero"
        name="0"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="grayBtn"
        name="."
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="orangeBtn"
        name="="
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
